flink-yarn/src/main/java/org/apache/flink/yarn/YarnFlinkResourceManager.java

```diff
@@ -335,8 +335,7 @@ protected void requestNewWorkers(int numWorkers) {
 			Priority priority = Priority.newInstance(0);
 
 			// Resource requirements for worker containers
-			int taskManagerSlots = taskManagerParameters.numSlots();
-			int vcores = config.getInteger(ConfigConstants.YARN_VCORES, Math.max(taskManagerSlots, 1));
+			int vcores = Math.max(taskManagerParameters.numSlots(), 1);
```

Comment:
I think we should still keep the possibility open to configure the `vcores` explicitly.